docs/source/tutorial/multi_gpu_vanilla.rst

```diff
@@ -0,0 +1,323 @@
+                    x = x.relu_()
+                xs.append(x.cpu())
+                pbar.update(1)
+            x_all = torch.cat(xs, dim=0)
+
+        pbar.close()
+        return x_all
+
+Defining our Spawnable Runner
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+   import os
+   import torch.distributed as dist
+   def run(rank, world_size, dataset):
+       os.environ['MASTER_ADDR'] = 'localhost'
+       os.environ['MASTER_PORT'] = '12355'
+       dist.init_process_group('nccl', rank=rank, world_size=world_size)
+
+       data = dataset[0]
+       data = data.to(rank, 'x', 'y')
+
+Notice we move to features and labels to GPU for faster feature fetching.
+
+Now we split training indices into :obj:`world_size` many chunks for each GPU:
+
+.. code-block:: python
+
+       train_idx = data.train_mask.nonzero(as_tuple=False).view(-1)
+       train_idx = train_idx.split(train_idx.size(0) // world_size)[rank]
+
+       kwargs = dict(batch_size=1024, num_workers=4, persistent_workers=True)
+       train_loader = NeighborLoader(data, input_nodes=train_idx,
+                                     num_neighbors=[25, 10], shuffle=True,
+                                     drop_last=True, **kwargs)
+
+We also create a single-hop evaluation neighbor loader:
+
+.. code-block:: python
+
+       if rank == 0:
+           subgraph_loader = NeighborLoader(copy.copy(data), num_neighbors=[-1],
+                                            shuffle=False, **kwargs)
+           # No need to maintain these features during evaluation:
+           del subgraph_loader.data.x, subgraph_loader.data.y
+           # Add global node index information:
+           subgraph_loader.data.node_id = torch.arange(data.num_nodes)
+
+Now that we have our data loaders defined initialize our model and wrap it in DistributedDataParallel
+
+.. code-block:: python
+
+      from torch.nn.parallel import DistributedDataParallel
+      torch.manual_seed(12345)
+      model = SAGE(dataset.num_features, 256, dataset.num_classes).to(rank)
+      model = DistributedDataParallel(model, device_ids=[rank])
+
+Now we set up our optimizer and define our training loop. Notice that we move the edge indices of each mini batch to GPU while the features and labels are already on GPU.
+
+.. code-block:: python
+
+      optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
+
+      for epoch in range(1, 21):
+        model.train()
+        for batch in train_loader:
+            optimizer.zero_grad()
+            out = model(batch.x, batch.edge_index.to(rank))[:batch.batch_size]
+            loss = F.cross_entropy(out, batch.y[:batch.batch_size])
+            loss.backward()
+            optimizer.step()
+
+After each training epoch, we evaluate and report accuracies:
+
+.. code-block:: python
+
+        dist.barrier()
+
+        if rank == 0:
+            print(f'Epoch: {epoch:02d}, Loss: {loss:.4f}')
+
+        if rank == 0 and epoch % 5 == 0:  # We evaluate on a single GPU for now
```

Comment:
I think it is important to mention here why we only perform this on `rank == 0`.